marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -149,25 +148,24 @@ handleQuery (ReceiveWebSocketMessage msg next) = do
                 Left decodingError -> addToast $ decodingErrorToast "Failed to parse contract update." decodingError
                 Right contractHistory@{ chParams, chHistory } ->
                   for_ chParams \(marloweParams /\ marloweData) -> do
+                    let
+                      walletDetails = view _walletDetails playState
                     currentSlot <- use _currentSlot
                     case lookup plutusAppId (view _allContracts playState) of
                       Just contractState -> do
                         selectedStep <- peruse $ _playState <<< _selectedContract <<< _selectedStep
-                        modifying (_playState <<< _allContracts) $ insert plutusAppId $ Contract.updateState currentSlot chHistory contractState
+                        modifying (_playState <<< _allContracts) $ insert plutusAppId $ Contract.updateState walletDetails marloweParams currentSlot chHistory contractState
```

Comment:
I'd like to rename `updateState` for something less "generic", but I'm not sure at the moment what could be a good name 🤔

Reply:
hmm, yeah, me neither